### PR TITLE
353: Point refactoring

### PR DESCRIPTION
**Pull request by pimotte**
今日、両親の自動車を運転しました。= Today, I drove my parents car.

This PR is dependent on #83

### CODE REVIEW

**Comment by MrHug** [comment]
Looks good, but I want the fixes in #83 first... Voting in favour of this however, means voting in favour of #83. Thus let me do it like this

**Comment by MrHug** [comment]
IF I vote in favour of #83 THEN :+1:

**Comment by pimotte** [comment]
I'm pretty sure you just violated the Dependency Inversion Principle ;)

**Comment by MrHug** [comment]
Well yeah, but meh :P

**Comment by pimotte** [comment]
As per 349, I'm issuing a warning.

File: README.md [hunk 1]
```diff
@@ -416,12 +416,31 @@ One turn consists of proposing one rule-change and having it voted on.
...
+
+A rule implementing a point-awardable condition must at least specifiy:
+
+- The condition under which points may be awarded
+- How many points may be awarded
+
+The number of points for a point-awardable condition will be awarded upon merging
+of the PR that fulfills the condition, if the PR reflects this awarding of points.
+
+Players cannot obtain points using a point-awardable condition introduced that PR.
```
Comment (jdonkervliet) [inline README.md]: introduced _in_ that PR.

File: README.md [hunk 2]
```diff
@@ -416,12 +416,31 @@ One turn consists of proposing one rule-change and having it voted on.
...
+
+Upon request of another player, a the initiator of a PR must state how many points their PR adds
```
Comment (jdonkervliet) [inline README.md]: -a

**Comment by pimotte** [comment]
The typo's that @jdonkervliet mentioned where already fixed in the parent PR, but I rebased to reflect this. This means all votes here are invalid.

**Comment by MrHug** [comment]
:+1:

**Comment by MrHug** [comment]
I just spotted an oversight. A warning is not reset after a change...

Thus the following is possible;

Make a PR, immediate issue warning.
Change PR changes to "Rule x: I get 100000 points" when 71 hours 59 minutes have passed
1/1 votes are in favour (no matter what, because all other votes are voided and the warning is not reset)
Rule gets merged

Perhaps we should fix that...

**Comment by pimotte** [comment]
It would be nice to make that explicit, but I would hold that an action of such form would still violate 106.

**Comment by pimotte** [comment]
Since warnings are not currently reset, I'm going to say this is 2/2 and merge it.

**Pull request merged**